### PR TITLE
[ASCII-749] Add WithIntakeHostname

### PR DESCRIPTION
**Pull request by pgimalac**
What does this PR do?
---------------------
Add a `WithIntakeHostname` agent option.

Which scenarios this will impact?
-------------------
Agent using a Fakeintake, although the change should be transparent (backward compatible).

Motivation
----------
Use a custom intake name in a test, without copy pasting the implementation of `WithFakeintake`.

Additional Notes
----------------
Implementation is exactly the same as `WithFakeintake`.

### CODE REVIEW

**Comment by pducolin** [comment]
**❓ question**
Would you use it to test against an intake other than the fakeintake ? Would you use one real DD intake ?

**Review by pducolin — APPROVED** [review]

File: components/datadog/agentparams/params.go [hunk 1]
```diff
@@ -218,12 +218,24 @@ container_lifecycle.logs_no_ssl: true
...
 sbom.logs_dd_url: %[1]s:80
-sbom.logs_no_ssl: true`, fakeintake.Host)
+sbom.logs_no_ssl: true`, hostname)
 		p.ExtraAgentConfig = append(p.ExtraAgentConfig, extraConfig)
 		return nil
 	}
 }
 
+// WithIntakeName configures the agent to use the given hostname as intake
+// This option is incompatible with `WithFakeintake`.
```
Comment (pducolin) [inline components/datadog/agentparams/params.go:228]:
```suggestion
// This option is overwritten by `WithFakeintake`.
```

File: components/datadog/agentparams/params.go [hunk 2]
```diff
@@ -218,12 +218,24 @@ container_lifecycle.logs_no_ssl: true
...
+func WithIntakeHostname(hostname string) func(*Params) error {
+	return withIntakeHostname(pulumi.String(hostname))
+}
+
+// WithFakeintake installs the fake intake and configures the Agent to use it.
+// This option is incompatible with `WithIntakeHostname`.
```
Comment (pducolin) [inline components/datadog/agentparams/params.go:234]:
```suggestion
// This option is overwritten by `WithIntakeHostname`.
```

**Comment by pgimalac** [comment]
> Would you use it to test against an intake other than the fakeintake ? Would you use one real DD intake ?

As discussed IRL this is used when you need a custom intake name when playing with DNS.
I guess it could also be used for a real DD intake, but the `site` setting handles that.

**Pull request merged**